src/blockchain/log-processors/profit-loss/frozen-funds.ts

```diff
@@ -0,0 +1,87 @@
+export interface Trade extends
+  Pick<MarketsRow<BigNumber>, "minPrice" | "maxPrice">, // from market to which this trade belongs
+  Pick<TradesRow<BigNumber>, "price" | "numCreatorTokens" | "numCreatorShares" | "numFillerTokens" | "numFillerShares"> { // data associated with this Trade
+  longOrShort: "long" | "short"; // "long" if the user was long on this trade (ie. created a buy order, or filled a sell order). "short" if user was short on this trade (ie. created a sell order, or filled a buy order)
+  creatorOrFiller: "creator" | "filler"; // "creator" if the user was the creator of the Order to which this Trade belongs. "filler" if the user filled another creator's Order
+  realizedProfit: BigNumber; // denominated in tokens (eg. ETH). Profit which the user realized by executing this trade
+}
+
+export interface FrozenFundsParams {
+  frozenFundsBeforeEvent: FrozenFunds; // FrozenFunds prior to processing this FrozenFundsEvent
+  event: FrozenFundsEvent; // FrozenFundsEvent to process and return updated FrozenFunds as impacted by this event
+}
+
+// getFrozenFundsAfterEventForOneOutcome computes the next frozen funds for
+// a market outcome, using the passed current frozen funds and event causing
+// the frozen funds to be updated. getFrozenFundsAfterEventForOneOutcome owns
+// the authoritative business definition of how frozen funds are calculated.
+export function getFrozenFundsAfterEventForOneOutcome(params: FrozenFundsParams): FrozenFunds {
+  if (params.event === "ClaimProceeds") {
+    return {
+      // When a user claims market proceeds, they are (by
+      // definition) withdrawing to their wallet all tokens they have
+      // escrowed in this market, so we set frozen funds to zero.
+      frozenFunds: ZERO,
+    };
+  }
+
+  // params.event is a Trade executed by this user
+  const trade = params.event;
+  let frozenFundsAfterEvent = params.frozenFundsBeforeEvent.frozenFunds;
+
+  // Idea here is that frozen funds are defined by tokens
+  // sent/received, so we have to transform trade details into
+  // the actual quantity of tokens the user sent or received.
+  const mySharesSent = trade.creatorOrFiller === "creator" ? trade.numCreatorShares : trade.numFillerShares;
+  const tokensReceivedPrice = trade.longOrShort === "long" ? trade.price : trade.maxPrice.minus(trade.price);
+  const myTokensReceived = mySharesSent.multipliedBy(tokensReceivedPrice);
+  const myTokensSent = trade.creatorOrFiller === "creator" ? trade.numCreatorTokens : trade.numFillerTokens;
+
+  // Tokens received are subtracted from frozen funds because the
+  // user now possesses those funds, and thus those funds are no longer
+  // "frozen". Similarly, tokens sent are added to frozen funds.
+  frozenFundsAfterEvent = frozenFundsAfterEvent.minus(myTokensReceived);
+  frozenFundsAfterEvent = frozenFundsAfterEvent.plus(myTokensSent);
+
+  if (frozenFundsAfterEvent.isGreaterThanOrEqualTo(
```

Comment:
As I mentioned I suspect we should actually remove this condition and just apply profit any time there if a non-zero PL realized. Tests should reveal this is correct once they've been added